### PR TITLE
22Q2-44 ✅ Replace networkingv1beta1 with networkingv1 for Ingress

### PR DESCRIPTION
**Pull request by yngvark**
## Description

Replace ingress version networkingv1beta1 with networkingv1.

## Motivation and Context

[22Q2-44](https://trello.com/c/POlaUKOa/44-bytte-fra-ingress-ingress-networkingv1beta1-til-v1)

## How to prove the effect of this PR?

* Run `okctl apply application` on an existing or a new application.
* Inspect `ingress.yaml` in the application's directory.
* The version should be networkingv1.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I have updated the release notes (for the next release).

### CODE REVIEW

File: pkg/scaffold/resources/ingress.go [hunk 1]
```diff
@@ -24,61 +23,77 @@ func CreateOkctlIngress(app v1alpha1.Application) (networkingv1beta1.Ingress, er
...
 	}
 
 	ingress := generateDefaultIngress()
 	ingress.ObjectMeta.Namespace = app.Metadata.Namespace
 
 	ingress.ObjectMeta.Name = app.Metadata.Name
 
-	ingress.Spec.Rules = []networkingv1beta1.IngressRule{
+	// https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/tasks/ssl_redirect/
+	pathType := networkingv1.PathTypeImplementationSpecific
```
Comment (deifyed) [inline pkg/scaffold/resources/ingress.go:79]: Are you sure `ImplementationSpecific` path type suits non-https-redirect routes?

ref: https://kubernetes.io/docs/concepts/services-networking/ingress/#path-types

I'd assume `Prefix` was the most natural choice

**Review by deifyed — APPROVED** [review]
Approved when my comment regarding PathType is answered. 

We should also remember that while our apps might support ingressv1, some of our packaged components does not. An example is kubepromstack

File: pkg/scaffold/resources/ingress.go (hunk 1 above)
Comment (yngvark) [reply pkg/scaffold/resources/ingress.go:79]: To be honest, I am not sure, but I am going to test it. I first used Prefix, but then I saw the AWS specific Ingress example using PathTypeImplementationSpecific (see doc in line 78).

Although that example use

```
 alb.ingress.kubernetes.io/ssl-redirect
```

while we use

```
alb.ingress.kubernetes.io/actions.ssl-redirect
```

so I need to dig a bit more into this.

Also, the link you provided says

> Implementations can treat this as a separate pathType or treat it identically to Prefix or Exact path types.

which _can_ mean that the ALB ingress controller can choose to treat it like Prefix, so I was thinking the ALB ingress controller knows best. But yeah, I'll more into it.

Comment (deifyed) [reply pkg/scaffold/resources/ingress.go:79]: I'm sure implementation specific is correct for the ssl redirect route, I just assumed prefix was the natural one for when you didn't want to handle routing somewhere else. Super interested in the result of your research

**Comment by yngvark** [comment]
Closing pull request as we won't be working with this any time soon.

**Pull request closed without merging**